pkg/distributor/query.go

```diff
@@ -29,6 +29,12 @@ import (
 	"github.com/grafana/mimir/pkg/util/spanlogger"
 )
 
+var (
+	// readActive is a ring.Operation that only selects instances marked as ring.ACTIVE, and
+	// extends the replica set in the same cases ring.Write would.
+	readActive = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, ring.Write.ShouldExtendReplicaSetOnState)
```

Comment:
the distributor does writes without extending the replication set. Why does the querier need to extend the replication set on failures? https://github.com/grafana/mimir/blob/184590b25f910f9ed0a543251a1ee26c36fc465c/pkg/distributor/distributor.go#L1140

Reply:
It doesn't - I should have read the distributor code rather than assuming it used `ring.Write` 😄 

Given distributors use `ring.WriteNoExtend`, do you reckon it's clearer for queriers to use the `ring.WriteNoExtend`, or to create another `Operation` with the same settings? Using `ring.WriteNoExtend` would make it clear that we're trying to match what's used for writes (and ensure they stay in sync), whereas creating a separate `Operation` instance would remove the confusion about using something with "write" in the name from something that does reads.

Reply:
My take on this topic:
- Operations shouldn't be defined in `dskit` at all. Operations are domain specific, and what we call `ring.WriteNoExtend` is actually a `ingester.WriteNoExtend`. In Mimir, excluding the ingesters ring, all other components define their own operations in their own package.
- To keep consistency with the rest of the code (where we define both "read" and "write" operations), and also to have better clarity when reading the code, I suggest to define the read operation too, and not use the write one on the read path (even if they're the same).

Reply:
I think in this case the read and write operations happen to be identical, but they are separate operations and may have separate semantics. So I agree having them separate makes more sense.

Reply:
Makes sense, I've made this change in 92c096a